src/components/Layout/Layout.js

```diff
@@ -183,11 +214,22 @@ const Layout = (props) => {
               </Grid>
             )}
 
-            {(value === "Home" ||
-              value === "Search" ||
-              value === "Profile" ||
-              value === "Notifications" ||
-              value === "Bookmarks") && (
+            {([
+              "Home",
+              "Search",
+              "Profile",
+              "Notifications",
+              "Bookmarks",
+            ].includes(value) ||
+              [
+                "/layout/home",
+                "/layout/profile",
+                "layout/notifications",
+                "/layout/explore",
+                "/layout/explore",
+                "/layout/search",
+                "/layout/bookmarks",
+              ].includes(window.location.pathname)) && (
```

Comment:
Looks too hacky. Maybe you should read a little bit more about react-router-dom to be able to handle this better.
Generally, you don't need window.location(that's a native implementation) if you've got react-router-dom which's a wrapper. 

Maybe why the pages don't work well without clicking items in the sidebar is because **useEffect** is required. Just check in case that's what you're trying to solve.